registry/txt.go

```diff
@@ -34,6 +35,26 @@ const (
 	providerSpecificForceUpdate = "txt/force-update"
 )
 
+type TXTFormat string
+
+const (
+	TXTFormatTransition   TXTFormat = "transition"
+	TXTFormatOnlyMetadata TXTFormat = "only-metadata"
+)
+
+func ParseTXTFormat(name string) TXTFormat {
+	switch name {
+	case TXTFormatTransition.String():
+		return TXTFormatTransition
+	default:
```

Comment:
so if we do things wrong we hide errors by choosing `TXTFormatOnlyMetadata`, IMO wrong choice.
Be explicit and have an UknownFormat for example in the enum that is the 0-value.